src/lib/y2partitioner/dialogs/bcache.rb

```diff
@@ -158,76 +233,79 @@ def validate
           end
         end
 
-        # returns device selected in widget. Only when dialog succeed and store is called.
-        # Otherwise undefined
-        attr_reader :result
-      end
+      private
 
-      # Widget to select the caching device
-      class CachingDevice < CWM::ComboBox
-        # @param device [Y2Storage::BcacheCset,nil] existing caching device or nil if it is a new bcache
-        # @param devices [Array<Y2Storage::BlkDevice,Y2Storage::BcacheCset>] possible devices that can be
-        #   used as caching device for the Bcache
-        def initialize(device, devices)
-          textdomain "storage"
-          @device = device
-          @devices = devices
+        # When the bcache exists, its backing device should be the default device.
+        # Otherwise, the first available device is the default one.
+        def default_device
+          return bcache.backing_device if bcache
+
+          super
         end
+      end
 
+      # Widget to select the caching device
+      class CachingDeviceSelector < DeviceSelector
         # @macro seeAbstractWidget
         def label
           _("Caching Device")
         end
 
         # @macro seeItemsSelection
         def items
-          @devices.map do |dev|
-            case dev
-            when Y2Storage::BcacheCset
-              [dev.sid.to_s, dev.display_name]
-            else
-              [dev.sid.to_s, dev.name]
-            end
-          end
+          items = devices.map { |d| item_for_device(d) }
+
+          items.prepend(item_for_non_device)
         end
 
         # @macro seeAbstractWidget
         def help
-          "<p>" +
-            # TRANSLATORS: %s stands for name of option
-            format(_(
-                     "%s is the device that will be used as caching device for bcache." \
-                     "It should be faster and usually is smaller than the backing device, " \
-                     "but it is not required. " \
-                     "The device will be formatted so any previous content will be wiped out."
-            ), "<b>" + label + "</b>") +
-            "</p>"
+          # TRANSLATORS: %{label} is replaced by the label of the option to select a caching device.
+          format(
+            _("<p>" \
+                "<b>%{label}</b> is the device that will be used as caching device for bcache." \
+                "It should be faster and usually is smaller than the backing device, " \
+                "but it is not required. The device will be formatted so any previous " \
```

Comment:
"it" or "that"? Already discussed offline

Reply:
Due to we have now a new paragraph to explain that the bcache can be created without caching, I would use "that" here. BTW, I would say "but that is not strictly required".